esi_ui/api/esi_api.py

```diff
@@ -116,6 +91,25 @@ def node_list_full_info(connection):
     return node_infos
 
 
+def set_provision_state(request, node):
```

Comment:
I think it may be better to simply split this function up into 'deploy' and 'undeploy'; `set_provision_state` is an existing Ironic node function that doesn't necessarily deploy or undeploy.